WordPress/openverse-catalog
# Refactor Brooklyn Museum to use ProviderDataIngester

## Fixes


Fixes #583 by @stacimc

## Description


This PR refactors the Brooklyn Museum provider API script to use the provider data ingester. The changes here are straightforward, mostly just moving things into static methods and removing the extra machinery that's covered by the provider data ingester base class.

I also consolidated the tests a bit so we didn't have to repeat the test cases.

As part of this change, I made a change to the `ProviderDataIngester` base class's `get_response_json` method. Since this function is a wrapper around the delayed requester, it felt easier to augment this than to access the requester directly. This accomplishes a similar task to what @obulat did in https://github.com/WordPress/openverse-catalog/pull/630#discussion_r950515494. i also removed the `typing` module from the `provider_data_ingester.py` module so that subclasses don't also need to import `typing`. I hope these changes are okay, but let me know if you have any reservations @stacimc!

## Testing Instructions

`just test` should work okay, unfortunately we don't have an API key (see #385 :see_no_evil:) so we can't _actually_ run this DAG. I've requested one so hopefully we can address that soon!

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
